conjur/controller/init_controller.py

```diff
@@ -97,23 +98,28 @@ def get_server_certificate(self):
 
         return fetched_certificate
 
+    # pylint: disable=line-too-long,logging-fstring-interpolation,broad-except,raise-missing-from
     def get_account_info(self, conjurrc_data):
         """
         Method to fetch the account from the user
         """
         if conjurrc_data.account is None:
             try:
                 self.init_logic.fetch_account_from_server(self.conjurrc_data)
-            # pylint: disable=broad-except,logging-fstring-interpolation
+            except CertificateHostnameMismatchException:
+                raise
             except Exception as error:
-                # pylint: disable=line-too-long,logging-fstring-interpolation
-                logging.warning(f"Unable to fetch the account from the Conjur server. Reason: {error}")
-                # If there was a problem fetching the account from the server, we will request one
-                conjurrc_data.account = input("Enter the Conjur account name (required): ").strip()
-
-                if conjurrc_data.account is None or conjurrc_data.account == '':
-                    # pylint: disable=raise-missing-from
-                    raise RuntimeError("Error: account is required")
+                # Check for catching if the endpoint is exists. If the endpoint does not exist,
+                # a 401 status code will be returned.
+                # If the endpoint does not exist, the user will be prompted to enter in their account.
+                # pylint: disable=no-member
+                if hasattr(error.response, 'status_code') and error.response.status_code == 401:
```

Comment:
@mbenita-Cyberark @tzheleznyak is there a better way to do this?
Not all exceptions have status codes and for those that do I want to check that the status code == 401 because in Conjur that means the endpoint doesnt exist. If the endpoint does exist it means we are interacting with Conjur OSS and want to ask for the account. All other errors, we want to fail and quit the process.

Reply:
No problam with the way you checked, but if it is not an http_error and status_code could be a string then you might get a runtime error. so add also `and isinstance( error.response.status_code, (int,float))`. 
maybe you can check if error is typeof httpError and it will be more easier.

but few more on the logic
1) 401 can be thrown in other cases as well. in that case the user will get a misleading error message. is there a way to be sure the account name is missing? 
2) I would take this logic to a private function called `handle_bad_input()`. it will be clearer in the code

Reply:
>                 if hasattr(error.response, 'status_code') and error.response.status_code == 401:

This check should only be true in OSS. I know that if this error is returned we are dealing with an OSS server and therefore need to ask the user for the account.